src/Cli/dotnet/commands/dotnet-workload/install/IInstaller.cs

```diff
@@ -20,7 +20,13 @@ internal interface IInstaller : IWorkloadManifestInstaller
 
         void RepairWorkloads(IEnumerable<WorkloadId> workloadIds, SdkFeatureBand sdkFeatureBand, DirectoryPath? offlineCache = null);
 
-        void GarbageCollectInstalledWorkloadPacks(DirectoryPath? offlineCache = null);
+        void GarbageCollectInstalledWorkloadPacks(DirectoryPath? offlineCache = null, bool cleanAllPacks = false);
+
+        /// <summary>
+        /// Remove all non-VS workload installation records for the specified feature band, and all records for feature bands lower than featureBand.
+        /// </summary>
+        /// <param name="featureBand">The maximum feature band to remove all workload installation records from.</param>
```

Comment:
Maybe rename the parameter to `maxFeatureBand` or something that indicates an upper bound.

Reply:
Or `currentFeatureBand`.  Also for the file-based installer I think we could go ahead and delete all of the records regardless of band, so we might not need the parameter at all in that case.